File: javascript/high-scores/high-scores.js
export class HighScores {
  constructor(scores) {
    this._scores = scores;
    this._scores_sorted_descending = [...scores].sort( (a, b) => {
      return b - a;
    });
  }

  get scores() {
    return this._scores;
  }

  get latest() {
    return this._scores[this._scores.length - 1];
  }

  get personalBest() {
    return this._scores_sorted_descending[0];
  }

  get personalTopThree() {
    return this._scores_sorted_descending.slice(0, 3);
  }
}

File: csharp/high-scores/HighScores.cs
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private readonly List<int> ScoreList;

    public HighScores(List<int> list) => ScoreList = list;

    public List<int> Scores() =>
        // Since the usage of this API is unknown, returns a copy 
        // to prevent external changes to internal state.
        ScoreList.ToList();

    public int Latest() => ScoreList.Last();

    public int PersonalBest() => ScoreList.Max();

    public List<int> PersonalTopThree() 
        => ScoreList.OrderByDescending(v => v)
                    .Take(3)
                    .ToList();
}